main.go

```diff
@@ -128,9 +133,9 @@ func main() {
 		time.Duration(*resyncInterval)*time.Second,
 		factoryOpts...)
 	applicationController := sparkapplication.NewController(
-		crdClient, kubeClient, apiExtensionsClient, factory, *submissionRunnerThreads, metricConfig, *namespace)
+		crdClient, kubeClient, apiExtensionsClient, factory, *submissionRunnerThreads, appDefaultConfig, metricConfig, *namespace)
```

Comment:
@liyinan926 I also thought about merging metricConfig and appDefaultConfig - as they are usually representing what wass passed as argument to operator

Reply:
I think we can bundle everything after `factory` together into a single struct named `OperatorConfig`.